### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by doliere** [microsoft/playwright#20347]
**Context:**
- Playwright Version: 1.29.2
- Operating System: Mac
- Node.js version: v18.11.0
- Browser: Chromium
- Extra: 

**Code Snippet**

Help us help you! Put down a short code snippet that illustrates your bug and
that we can run and debug locally. For example:

```javascript
const playwright = require('playwright');

(async () => {
    let browser = await playwright.chromium.launch({
        headless: false
    })
    let page = await browser.newPage();
    await page.goto('https://easyimages.net');
}) ();
```

**Describe the bug**

The browser keeps crashing with the following error
```javascript
/Users/johndoe/node_modules/playwright-core/lib/server/chromium/crExecutionContext.js:153
    return '[' + String(result) + ']';
                 ^

RangeError: Invalid string length
    at Array.join (<anonymous>)
    at Array.toString (<anonymous>)
    at String (<anonymous>)
    at renderPreview (/Users/johndoe/node_modules/playwright-core/lib/server/chromium/crExecutionContext.js:153:18)
    at CRExecutionContext.createHandle (/Users/johndoe/node_modules/playwright-core/lib/server/chromium/crExecutionContext.js:108:80)
    at FrameExecutionContext.createHandle (/Users/johndoe/node_modules/playwright-core/lib/server/javascript.js:67:27)
    at FrameExecutionContext.createHandle (/Users/johndoe/node_modules/playwright-core/lib/server/dom.js:84:18)
    at /Users/johndoe/node_modules/playwright-core/lib/server/chromium/crPage.js:692:50
    at Array.map (<anonymous>)
    at FrameSession._onConsoleAPI (/Users/johndoe/node_modules/playwright-core/lib/server/chromium/crPage.js:692:31)
```

I have tried the same code with Puppeteer and the issue was not there. The problem seems to be related to iframes that are loaded within the particular site.

**Comment by pavelfeldman** [comment]
I can repro, this looks like an OOM while building the console object previews. Not sure how Puppeteer is relevant, but this looks like a real Playwright bug.

**Comment by doliere** [comment]
I upgraded to Playwright 1.30.0, but I can still reproduce the bug, on Mac, Linux

**Comment by aslushnikov** [comment]
@doliere the fix didn't make it to 1.30 - it'll be released in 1.31. You can try it out on Playwright canary:

```
npm i @playwright/test@next
```

**Comment by doliere** [comment]
that works, thanks :-)
